### ISSUE TITLE
Error Phalcon DevTools Models Oracle

### ISSUE THREAD

**Issue by verde3ro** [phalcon/phalcon-devtools#1274]
I have the following problem when generating the models of an Oracle database.

```
phalcon create-all-models --mapcolumn --doc --force --schema=MGR_ARVYC

Phalcon DevTools (3.4.0)

ERROR: SQLSTATE[HY000]: General error: 24337 OCIStmtExecute: ORA-24337: statement handle not prepared (ext\pdo_oci\oci_statement.c:159)
```

I hope you can help me.

**Comment by Jeckerson** [comment]
@verde3ro Can you provide example models when it occurs?

**Comment by verde3ro** [comment]
Occurs when generating models with Phalcon DevTools or Web Tools.

![2019-03-19 08_50_03-Window](https://user-images.githubusercontent.com/13452905/54615753-3d588400-4a24-11e9-8687-2ff58a0a15c6.png)
![2019-03-19 08_50_44-Window](https://user-images.githubusercontent.com/13452905/54615755-3d588400-4a24-11e9-9719-5ea5a3b960c6.png)

**Comment by Jeckerson** [comment]
@verde3ro Please execute model generation per single table and provide output of `SHOW CREATE TABLE` of tables where it occurs.

**Comment by verde3ro** [comment]
@Jeckerson,  Execution generation of model per single table:

Table: 
```
CREATE TABLE "AV_USUARIO_C" 
   (	"ID" NUMBER DEFAULT NULL NOT NULL ENABLE, 
	"USUARIO" VARCHAR2(20 BYTE) DEFAULT '' NOT NULL ENABLE, 
	"PASSWORD" VARCHAR2(50 BYTE) DEFAULT '', 
	"NOMBRE" VARCHAR2(250 BYTE) DEFAULT '' NOT NULL ENABLE, 
	"CORREO" VARCHAR2(100 BYTE) DEFAULT '', 
	"ESTATUS" VARCHAR2(5 BYTE) DEFAULT 'AC' NOT NULL ENABLE, 
	"USUARIO_I" VARCHAR2(20 BYTE) DEFAULT '' NOT NULL ENABLE, 
	"FECHA_I" DATE DEFAULT SYSDATE NOT NULL ENABLE, 
	"USUARIO_U" VARCHAR2(20 BYTE), 
	"FECHA_U" DATE, 
	"CORRECTO" NUMBER(1,0) DEFAULT 0, 
	"USUARIO_EBS" VARCHAR2(20 BYTE) DEFAULT '', 
	 CONSTRAINT "AV_USUARIO_PK" PRIMARY KEY ("ID"), 
	 CONSTRAINT "AV_USUARIO_UX" UNIQUE ("USUARIO")
   ) ;
```

Configuration:

```
 'database' => [
        'adapter'     => 'Oracle',
        'host'        => '172.17.21.33:1521/XE',
        'username'    => 'MGR_ARVYC',
        'password'    => 'a4rvic',
        'dbname'      => '(DESCRIPTION=(ADDRESS_LIST = (ADDRESS = (PROTOCOL = TCP)(HOST = 172.17.21.33)(PORT = 1521))) (CONNECT_DATA=(SID=XE)))',
        'charset'     => 'utf8',
    ],
```
Execution:

![2019-03-19 08_50_44-Window](https://user-images.githubusercontent.com/13452905/54629136-8b798180-4a3c-11e9-9fa1-7e03ca44ae27.png)
![2019-03-19 11_38_27-Window](https://user-images.githubusercontent.com/13452905/54629137-8c121800-4a3c-11e9-9fad-03430bd4b953.png)
![2019-03-19 11_38_39-Window](https://user-images.githubusercontent.com/13452905/54629138-8c121800-4a3c-11e9-87fd-f0ea4daa64d0.png)

**Comment by Jeckerson** [comment]
`Option config hasn't been defined` is related to another task. That means that your table name will be generated after #1292 will be merged.

Try to find another table that generate this error:
`ERROR: SQLSTATE[HY000]: General error: 24337 OCIStmtExecute: ORA-24337: statement handle not prepared (ext\pdo_oci\oci_statement.c:159)`

**Comment by verde3ro** [comment]
I did not understand the comment:

> Option config hasn't been defined is related to another task. That means that your table name will be generated after #1292 will be merged.


Any table in the schema generate the error when created the model.

```
ERROR: SQLSTATE[HY000]: General error: 24337 OCIStmtExecute: ORA-24337: statement handle not prepared
 (ext\pdo_oci\oci_statement.c:159)
```

![2019-04-02 10_20_21-Error Phalcon DevTools Models Oracle · Issue #1274 · phalcon_phalcon-devtools](https://user-images.githubusercontent.com/13452905/55418972-06f52b80-5531-11e9-90c2-5321ebc37551.png)

**Comment by Jeckerson** [comment]
`I did not understand the comment:`

You posted screenshot where error is different
https://user-images.githubusercontent.com/13452905/54629138-8c121800-4a3c-11e9-87fd-f0ea4daa64d0.png

**Comment by Jeckerson** [comment]
So your issue is actual.

**Comment by verde3ro** [comment]
yes it's actual

**Comment by verde3ro** [comment]
```
ERROR: SQLSTATE[HY000]: General error: 24337 OCIStmtExecute: ORA-24337: statement handle not prepared
 (ext\pdo_oci\oci_statement.c:159)

```
![2019-04-02 10_20_21-Error Phalcon DevTools Models Oracle · Issue #1274 · phalcon_phalcon-devtools](https://user-images.githubusercontent.com/13452905/55420683-db744000-5534-11e9-9850-147736fe9a05.png)

**Comment by Jeckerson** [comment]
Can be fixed only when Incubator DB Component is ready for Phalcon v4.

https://github.com/phalcon/incubator-db

**Comment by flyingangel** [comment]
Error `Option namespace hasn't been defined` when generating a class with constraint relation.

There's 2 typos in the file Model.php

`if ($this->modelOptions->hasOption('namespace'))` instead of `if ($this->modelOptions->getOption('namespace'))`

Devtool tag 4.0.5

**Comment by Jeckerson** [comment]
@flyingangel I think your issue isn't related to this one. Please create separate one.
